lib/stack_master/parameter_resolvers/stack_output.rb

```diff
@@ -34,21 +34,27 @@ def cf
         @cf ||= StackMaster.cloud_formation_driver
       end
 
-      def validate_value!(value)
-        if !value.is_a?(String) || !value.include?('/')
-          raise ArgumentError, 'Stack output values must be in the form of stack-name/output-name'
+      def parse!(value)
+        if !value.is_a?(String) || !(match = @output_regex.match(value))
+          raise ArgumentError, 'Stack output values must be in the form of [region:]stack-name/output_name'
         end
+
+        match.captures
       end
 
-      def find_stack(stack_name)
-        @stacks.fetch(stack_name) do
+      def find_stack(stack_name, region)
+        stack_key = stack_key(stack_name, region)
+        @stacks.fetch(stack_key) do
+          original_region = cf.region
+          cf.set_region(region) if region && original_region != region
           cf_stack = cf.describe_stacks(stack_name: stack_name).stacks.first
-          @stacks[stack_name] = cf_stack
+          cf.set_region(original_region) if region && original_region != region
+          @stacks[stack_key] = cf_stack
         end
       end
 
-      def cf
-        @cf ||= StackMaster.cloud_formation_driver
+      def stack_key(stack_name, region)
+        "#{region || 'any'}:#{stack_name}"
```

Comment:
If the logic is changed earlier in this class, then there's no need for `|| 'any'` as `region` will never be `nil`.